### ISSUE TITLE
smallrye-openapi extension exposing rest client interface

### ISSUE THREAD

**Issue by edeandrea** [quarkusio/quarkus#20388]
### Describe the bug

When building an application that uses the `smallrye-openapi` extension and also registering a rest client via `@RegisterRestClient`, when opening up the swagger-ui page looking at the openapi document, the methods declared in the rest client interface show up.

### Expected behavior

The rest client interface and the methods declared in them should not show up in the swagger-ui page nor the openapi document.

### Actual behavior

The methods declared in the rest client interface show up.

![Screen Shot 2021-09-24 at 2 29 56 PM](https://user-images.githubusercontent.com/363447/134723573-8d6197da-2e48-4837-b322-d899593f6919.png)

### How to Reproduce?

1. Clone https://github.com/edeandrea/fruity-service
2. Run dev mode
3. Open browser to http://localhost:8080/q/swagger-ui
    - The `/api/fruit/{name}` entry should not be there
4. Open browser to http://localhost:8080/q/openapi
    - The `/api/fruit/{name}` entry should not be there

In the repo in step 1, only the methods in `src/main/java/org/acme/rest/FruitResource.java` should be there. The methods defined in `src/main/java/org/acme/service/FruityViceClient.java` should not be there.

### Output of `uname -a` or `ver`

```
Darwin edeandre-mac 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64
```

### Output of `java -version`

```
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

### Quarkus version or git rev

`2.2.3.Final`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.1-bin/2l5mhf2pq2clrde7f7qp1rdt5m/apache-maven-3.8.1
Java version: 11.0.2, vendor: Oracle Corporation, runtime: /Users/edeandre/.sdkman/candidates/java/11.0.2-open
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```

**Comment by MikeEdgar** [comment]
Fixed in upstream smallrye/smallrye-open-api#919 and in release 2.1.15.

**Comment by phillip-kruger** [comment]
Yes this is already in main branch. Closing here
